src/api.ts

```diff
@@ -59,9 +268,41 @@ export function generateOptionsString(input: GetINCHIOptions | GetINCHIExOptions
   return options.join(" ");
 }
 
-export interface INCHIOutput {
-  szInChI: string;
-  szAuxInfo: string;
-  szMessage: string;
-  szLog: string;
+// #endregion
+
+// #region Public Functions
+
+export function CheckINCHIKey(input: string): CheckINCHIKeyReturnCode {
+  return INCHIAPI.CheckINCHIKey(input);
 }
+
+export function CheckINCHI(input: string, strict?: boolean): CheckINCHIReturnCode {
+  return INCHIAPI.CheckINCHI(input, strict === true ? 1 : 0);
+}
+
+export function GetStringLength(input: string): number {
+  return INCHIAPI.GetStringLength(input);
+}
+
+export function GetStructFromINCHI(input: string, options: GetINCHIOptions): GetINCHIReturnCode {
+  const inchiIn = { szInChI: input, szOptions: generateOptionsString(options) };
```

Comment:
Looks incorrect. You need to properly instantiate `inchiIn` via `new inchi_XYZ(...)`.

Same issue with `GetStructFromINCHIEx` and `GetStructFromStdINCHI`.

Reply:
solv

> 

> `corrected it sir`